src/views/List.jsx

```diff
@@ -1,14 +1,70 @@
+import { useState, useEffect } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	const [searchTerm, setSearchTerm] = useState('');
+	const [filteredData, setFilteredData] = useState(data);
+
+	const handleChange = (e) => {
+		const searchTermLocal = e.target.value;
+		//Sets searchTerm to the input value
+		setSearchTerm(searchTermLocal);
+	};
+
+	//useEffect triggered with change in searchTerm or data
+	useEffect(() => {
+		//Debounce function
+		const getItem = setTimeout(() => {
+			function searchItem(listItem) {
```

Comment:
Optional: The function definition here isn't strictly necessary. You could set `const filteredResults = data.filter((item) =>...` and shave a few lines of code, but otherwise this is functioning as intended.

Reply:
@MiliMade  What do think about replace the function with assigning data.filter to filteredResults like above
<img width="1189" alt="image" src="https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/121919044/d9bf2658-4b95-4e5b-8ae4-33fb5745447d">